grammar/grammar.peg

```diff
@@ -50,6 +54,39 @@ NotExpression <- "not" _ expr:NotExpression {
    return expr, nil
 }
 
+CollectionExpression <- operator:("all"/"any") _ selector:Selector _ "as" _ ident:(Identifier / "_") _? "{" _? expr:OrExpression _? "}" {
+   return c.onCollectionExpressionWithKey1(operator, selector, ident, "", expr)
```

Comment:
These generated methods should probably not be used as newer versions of the library may not generate them with the same names and break this code. 

When I implemented the syntax before I used more sub-expressions to allow sharing more code between the different forms of name bindings: https://github.com/hashicorp/go-bexpr/blob/213d8c2fe4480cba2fad47e9a778bf9db6c3650b/grammar.peg#L61-L91

The sub-expressions can also return values eliminating the need to do as much string matching within the body of an expression (see the any/all operator matching from that snippet above)

Reply:
Indeed, I was wondering why the grammar is more verbose than it needs to be that it is required because we are using a generator. 

I have now reused most of your grammar and it should be clean now.